go/roothash/api/history.go

```diff
@@ -4,9 +4,24 @@ import (
 	"context"
 
 	"github.com/oasisprotocol/oasis-core/go/common"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
 	"github.com/oasisprotocol/oasis-core/go/roothash/api/block"
 )
 
+// RoundResults contains information about how a particular round was executed by the consensus
+// layer.
+type RoundResults struct {
+	// Messages are the results of executing emitted runtime messages.
+	Messages []*MessageEvent `json:"messages,omitempty"`
+
+	// GoodComputeNodes are the public keys of compute nodes that positively contributed to the
+	// round by replicating the computation correctly.
+	GoodComputeNodes []signature.PublicKey `json:"good_compute_nodes,omitempty"`
```

Comment:
Wondering if there's a better alternative naming to use for nodes that submitted correct and invalid results. I thought about it, but don't have a better suggestion though :D

Reply:
Yeah, I was thinking about how to name this as well, but didn't come up with anything better. Tendermint uses the term "Byzantine Validators" for the bad ones.